devenv/lib/venv.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import annotations
+
+import os
+import shutil
+from typing import Optional
+
+from devenv import pythons
+from devenv.constants import bin_root
+from devenv.constants import venvs_root
+from devenv.lib import config
+from devenv.lib import fs
+from devenv.lib import proc
+
+VENV_OK = 1
+VENV_VERSION_MISMATCH = 2
+VENV_NOT_PRESENT = 3
+VENV_NOT_CONFIGURED = 4
```

Comment:
This is a prime case for using enums, it would enhance readability and reduce potential errors.

Reply:
What part of this isn't readable? I fail to see how using enums would be beneficial. If anything it's just more bloated.

Reply:
I understand where you are coming from, I prefer using enums for the semantics. Enums provide a clear indication that the values are part of a single group of related values, status codes in this case. This makes the code more self-documenting and readable in my opinion.

Reply:
alright, fair enough